src/librustc_traits/implied_outlives_bounds.rs

```diff
@@ -99,7 +99,8 @@ fn compute_implied_outlives_bounds<'tcx>(
                 | ty::Predicate::Projection(..)
                 | ty::Predicate::ClosureKind(..)
                 | ty::Predicate::ObjectSafe(..)
-                | ty::Predicate::ConstEvaluatable(..) => vec![],
+                | ty::Predicate::ConstEvaluatable(..)
+                | ty::Predicate::WellFormedConst(..) => vec![],
```

Comment:
I think this is fine because the only reason `wf::obligations` would return a `WellFormedConst` is for `ConstKind::Infer`, but, hmm. Why is `ty::Predicate::WellFormed` handled below?

It should also not be reachable, but I guess there are cases in which `wf::obligations` contains `WellFormed` obligations besides inference variables.

There's a comment above saying, but it makes no sense as typeck doesn't use this query:
```rust
        //                        (Note: normally there won't be
        // unresolved inference variables here anyway, but there might be
        // during typeck under some circumstances.)
```

cc @nikomatsakis Could you comment on this?